tests/istanbul/eip152_blake2/test_blake2.py

```diff
@@ -0,0 +1,501 @@
+"""
+abstract: Tests [EIP-152: BLAKE2 compression precompile](https://eips.ethereum.org/EIPS/eip-152)
+    Test cases for [EIP-152: BLAKE2 compression precompile](https://eips.ethereum.org/EIPS/eip-152).
+"""
+
+from dataclasses import dataclass
+
+import pytest
+
+from ethereum_test_tools import (
+    Account,
+    Alloc,
+    Environment,
+    StateTestFiller,
+    TestParameterGroup,
+    Transaction,
+    YulCompiler,
+)
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-152.md"
+REFERENCE_SPEC_VERSION = "5510973b40973b6aa774f04c9caba823c8ff8460"
+
+
+@dataclass(kw_only=True, frozen=True, repr=False)
+class Blake2Input(TestParameterGroup):
+    """
+    Helper class that defines the BLAKE2 precompile inputs and creates the
+    call data from them.
+
+    Attributes:
+        tx_data_rounds (bytes): The number of rounds for the BLAKE2 precompile.
+        h (str): The state vector for the BLAKE2 precompile.
+        m (str): The message block vector for the BLAKE2 precompile.
+        t_0 (str): Offset counter 0 for the BLAKE2 precompile.
+        t_1 (str): Offset counter 1 for the BLAKE2 precompile.
+        f (str): The final block indicator flag for the BLAKE2 precompile.
+
+    """
+
+    rounds: bytes
+    data: bytes = b""
+
+    def create_blake2_tx_data(self):
+        """Generate input for the BLAKE2 precompile."""
+        return self.rounds
+
+
+@dataclass(kw_only=True, frozen=True, repr=False)
+class ExpectedOutput(TestParameterGroup):
+    """
+    Expected test result.
+
+    Attributes:
+        returned_data (str): The output returnData is the expected output of the call
+
+    """
+
+    call_return_code: str
+    data_1: str
+    data_2: str
+
+
+@pytest.mark.valid_from("Istanbul")
+@pytest.mark.parametrize(
+    ["data", "output"],
+    [
+        pytest.param(
+            Blake2Input(rounds=b"\x00"),
```

Comment:
thats a bit ugly. do we have a class that makes Bytes from string ("0x112233") so that bytes are 11 22 33?
@marioevz

Reply:
That's a great point, I could actually just plug in the numbers as integers and convert those to bytes in the class itself. I think explicit numbers would make it really clear and easy to understand, would that be preferable to using hex values?